### ISSUE TITLE
Klarna Payment Provider

### ISSUE THREAD

**Issue by srindom** [medusajs/medusa#45]
We need to create a plugin that enables Klarna payments

**Comment by olivermrbl** [comment]
Klarna merchant_urls (webhooks):
- push: called when order has succeeded
- shipping option update: called on shipping option selection
- address update
Rest of the URLs are client side URLs and should be defined in `medusa-config.js`:
- terms: where you can read terms
- confirmation: where you are sent after payment is complete
- checkout: checkout flow

NOTE: Klarna payments must be completed in the local currency of the country. This is a restriction from Klarna

```js
createPayment(cart)
- createOrder(cart, merchant_urls)
- return data

updatePayment(payment_method.data, cart)
- updateOrder(order)
- return data

getStatus(payment_method.data)
- Status 1: Initial - created order but nothing else
- Status 2: Authorized - when checkout flow done
- Status 3: Succeeded || Canceled
- Status 4: Refunded
- return payment status

capturePayment(payment_method.data, order)
- total amount from order items
- createCapture(payment_method.data.klarna_id, amount)
- return data

refundPayment(payment_method.data, amount)
- createRefund(payment_method.data, amount)
- return data

deletePayment(payment_method.data)
- return intent
```

**Comment by olivermrbl** [comment]
Closed in #79
